orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/impl/OrientDBDatastoreProvider.java

```diff
@@ -46,58 +57,94 @@
 
 	@Override
 	public void start() {
-		LOG.info( "start" );
+		log.debug( "---start---" );
 		try {
 			PropertyReaderContext<String> jdbcUrlPropery = propertyReader.property( "javax.persistence.jdbc.url", String.class );
 			if ( jdbcUrlPropery != null ) {
-				String jdbcUrl = jdbcUrlPropery.getValue();
-				LOG.info( "jdbcUrl:" + jdbcUrl );
+				jdbcUrl = jdbcUrlPropery.getValue();
+				log.warn( "jdbcUrl:" + jdbcUrl );
 				Class.forName( propertyReader.property( "javax.persistence.jdbc.driver", String.class ).getValue() ).newInstance();
-				Properties info = new Properties();
+				info = new Properties();
 				info.put( "user", propertyReader.property( "javax.persistence.jdbc.user", String.class ).getValue() );
 				info.put( "password", propertyReader.property( "javax.persistence.jdbc.password", String.class ).getValue() );
-
-				connection = DriverManager.getConnection( jdbcUrl, info );
-
+				createInMemoryDB();
+				connectionHolder = new ConnectionHolder( jdbcUrl, info );
 			}
-
+			FormatterUtil.setDateFormater( new ThreadLocal<DateFormat>() {
+
+				@Override
+				protected DateFormat initialValue() {
+					SimpleDateFormat f = new SimpleDateFormat(
+							propertyReader.property( OrientDBProperties.DATE_FORMAT, String.class ).withDefault( "yyyy-MM-dd" ).getValue() );
+					return f;
+				}
+
+			} );
+			FormatterUtil.setDateTimeFormater( new ThreadLocal<DateFormat>() {
```

Comment:
Couldn't you create a Date type descriptor like you did for ORecord? Something similar to `CalendarTimeZoneDateTimeTypeDescriptor`